values.go

```diff
@@ -132,6 +135,61 @@ func (value *Number) Equal(that Value) bool {
 	return value.value == typed.value && value.typ.scale == typed.typ.scale
 }
 
+func (value *Number) GreaterThan(that Value) bool {
```

Comment:
Let's only have `GreaterThan` defined on `that *Number` similar to other operations like `Plus`.

In the `Binop` compute you can handle with non-numerical cases. You don't want to do this at the value level.